sources/packages/web/src/services/http/UserApi.ts

```diff
@@ -9,13 +9,33 @@ export default interface User {
 }
 
 export class UserApi extends HttpBaseClient {
-  public async checkUser(): Promise<User> {
+  public async checkUser(): Promise<boolean> {
+    let userExists = false;
     try {
       const response = await this.apiClient.get(
         "users/check-user",
         this.addAuthHeader(),
       );
-      return response.data as User;
+      if (response.status === 200) {
+        const user = response.data as User;
+        //TODO: This should not ideally rely on id.
```

Comment:
If the status code on API will be changed to "NOT FOUND", this check can be changed as you mentioned in the comments.